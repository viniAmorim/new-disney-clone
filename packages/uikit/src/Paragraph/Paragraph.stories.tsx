import React from 'react';

import { Story, Meta } from '@storybook/react';

import Paragraph, { ParagraphProps } from '~/Paragraph';

export default {
  title: 'Typography/Paragraph',
  component: Paragraph,
} as Meta;

const Template: Story<ParagraphProps> = (args) => <Paragraph {...args} />;

export const DisplayLight = Template.bind({});
DisplayLight.args = {
  children: 'Display',
  size: 'display',
};

export const HeadlineLight = Template.bind({});
HeadlineLight.args = {
  children: 'Headline',
  size: 'headline',
};

export const TitleLight = Template.bind({});
TitleLight.args = {
  children: 'Title',
  size: 'title',
};

export const SubheaderLight = Template.bind({});
SubheaderLight.args = {
  children: 'Subheader',
  size: 'subheader',
};

export const BodyLight = Template.bind({});
BodyLight.args = {
  children: 'Body',
  size: 'body',
};

export const SmallLight = Template.bind({});
SmallLight.args = {
  children: 'Small',
  size: 'small',
};
