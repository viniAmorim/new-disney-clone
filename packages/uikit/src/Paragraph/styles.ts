import styled, { css } from 'styled-components';

export type StyledParagraphProps = {
  size?: 'display' | 'headline' | 'title' | 'subheader' | 'body' | 'small';
};

const sizeModifier = {
  display: css`
    ${({ theme }) => theme.typography['display-light']}
    font-size: 36px;
    line-height: 48px;
  `,
  headline: css`
    ${({ theme }) => theme.typography['headline-light']}
    font-size: 24px;
    line-height: 32px;
  `,
  title: css`
    ${({ theme }) => theme.typography['title-light']}
    font-size: 20px;
    line-height: 28px;
  `,
  subheader: css`
    ${({ theme }) => theme.typography['subheader-light']}
    font-size: 16px;
    line-height: 24px;
  `,
  body: css`
    ${({ theme }) => theme.typography['body-default']}
    font-size: 14px;
    line-height: 20px;
  `,
  small: css`
    ${({ theme }) => theme.typography['small-light']}
    font-size: 12px;
    line-height: 16px;
  `,
};

export const Paragraph = styled.p<StyledParagraphProps>`
  ${({ size = 'display' }) => sizeModifier[size]}
`;
