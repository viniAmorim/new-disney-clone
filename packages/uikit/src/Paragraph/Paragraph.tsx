import React, { BaseHTMLAttributes, ReactNode } from 'react';

import * as Styled from './styles';

export interface ParagraphProps
  extends BaseHTMLAttributes<HTMLParagraphElement>,
    Styled.StyledParagraphProps {
  /**
   * Paragraph Content
   */
  children: ReactNode;
}

const Paragraph = ({ children, size, ...rest }: ParagraphProps) => {
  return (
    <Styled.Paragraph size={size} {...rest}>
      {children}
    </Styled.Paragraph>
  );
};

export default Paragraph;
