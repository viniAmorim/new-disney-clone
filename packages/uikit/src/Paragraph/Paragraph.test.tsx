import React from 'react';

import { render } from '~/test/utils';

import Paragraph from './Paragraph';

describe('Paragraph', () => {
  it('should render correctly', () => {
    const { getByText } = render(<Paragraph>Paragraph</Paragraph>);

    const textElement = getByText('Paragraph');

    expect(textElement).toBeInTheDocument();
    expect(textElement).toHaveTextContent(/^Paragraph$/);

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text display', () => {
    const { getByText } = render(
      <Paragraph size="display">Paragraph</Paragraph>,
    );

    const textElement = getByText('Paragraph');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text headline', () => {
    const { getByText } = render(
      <Paragraph size="headline">Paragraph</Paragraph>,
    );

    const textElement = getByText('Paragraph');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text title', () => {
    const { getByText } = render(<Paragraph size="title">Paragraph</Paragraph>);

    const textElement = getByText('Paragraph');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text subheader', () => {
    const { getByText } = render(
      <Paragraph size="subheader">Paragraph</Paragraph>,
    );

    const textElement = getByText('Paragraph');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text body', () => {
    const { getByText } = render(<Paragraph size="body">Paragraph</Paragraph>);

    const textElement = getByText('Paragraph');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text small', () => {
    const { getByText } = render(<Paragraph size="small">Paragraph</Paragraph>);

    const textElement = getByText('Paragraph');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });
});
