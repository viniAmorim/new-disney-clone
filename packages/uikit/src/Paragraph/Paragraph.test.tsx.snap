// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph should render a text body 1`] = `
<p
  class="sc-bdfBQB fElQMP"
>
  Paragraph
</p>
`;

exports[`Paragraph should render a text display 1`] = `
<p
  class="sc-bdfBQB tKTgq"
>
  Paragraph
</p>
`;

exports[`Paragraph should render a text headline 1`] = `
<p
  class="sc-bdfBQB cuiefd"
>
  Paragraph
</p>
`;

exports[`Paragraph should render a text small 1`] = `
<p
  class="sc-bdfBQB eWtOrH"
>
  Paragraph
</p>
`;

exports[`Paragraph should render a text subheader 1`] = `
<p
  class="sc-bdfBQB jhQjFt"
>
  Paragraph
</p>
`;

exports[`Paragraph should render a text title 1`] = `
<p
  class="sc-bdfBQB kzTOLQ"
>
  Paragraph
</p>
`;

exports[`Paragraph should render correctly 1`] = `
<p
  class="sc-bdfBQB tKTgq"
>
  Paragraph
</p>
`;
